### PR TITLE
Trivial unit test against the model and actual ChainDB

### PR DESCRIPTION
**Pull request by bartfrenk**
# Description

Adds a simple unit test for the StateMachine model of the ChainDB and the actual ChainDB.

Part of #4183 

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [x] The documentation has been properly updated
    - [x] New tests are added if needed and existing tests are updated
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

**Review by abailly-iohk — APPROVED** [review]
I like it :) 
This feels really similar to what qc-dynamic provides, but in a simpler and more focused way.

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs [hunk 1]
```diff
@@ -1593,6 +1604,8 @@ runCmdsLockstep maxClockSkew (SmallChunkInfo chunkInfo) cmds =
                            (remainingCleanups === 0)
       return (hist, prop)
 
+
```
Comment (dnadales) [inline ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs:1607]: Check extra spaces.

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                     unsafeNewRegistry)
+import           Test.Util.ChainDB (MinimalChainDbArgs (..), emptyNodeDBs,
+                     fromMinimalChainDbArgs, nodeDBsVol)
+
+import           Ouroboros.Consensus.Config (TopLevelConfig)
+import           Ouroboros.Consensus.Ledger.Extended (ExtLedgerState)
+import           Ouroboros.Consensus.Storage.ImmutableDB.Chunks as ImmutableDB
+
+tests :: TestTree
+tests = testGroup "trivial"
```
Comment (dnadales) [inline ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs:39]: Rename?

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+toAssertion :: Either TestFailure a -> Assertion
+toAssertion (Left (TestFailure t)) = assertFailure t
+toAssertion (Right _)              = pure ()
+
+assert :: (MonadError TestFailure m) => Bool -> m ()
+assert b = unless b $ throwError (TestFailure "boom")
+
+-- | Targets for the test expression need to instantiate this class.
+class Target m where
```
Comment (dnadales) [inline ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs:69]:
```suggestion
class SupportsUnitTests m where
```

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  newFollower
+    :: m (FollowerId m)
+  followerInstruction
+    :: FollowerId m -> m (Maybe (ChainUpdate (Block m) (AllComponents (Block m))))
+
+getFollowerInstruction :: (Target m, MonadError TestFailure m) => m ()
+getFollowerInstruction = do
+  f <- newFollower
+  instr <- followerInstruction f
+  assert $ isJust instr
```
Comment (dnadales) [inline ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs:82]: Maybe we can write the expectation. Something like:

```suggestion
  isJust instr `orErrorWithMsg` "Expecting Just value"
```

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    withChainDbEnv = withTestChainDbEnv topLevelConfig chunkInfo testInitExtLedger
+
+newtype TestFailure = TestFailure String deriving (Show)
+
+toAssertion :: Either TestFailure a -> Assertion
+toAssertion (Left (TestFailure t)) = assertFailure t
+toAssertion (Right _)              = pure ()
+
+assert :: (MonadError TestFailure m) => Bool -> m ()
+assert b = unless b $ throwError (TestFailure "boom")
```
Comment (dnadales) [inline ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs:66]: See suggestion about exiting with a specific error message.

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  newFollower = do
+    model <- get
+    let (followerId, model') = Model.newFollower model
+    put model'
+    pure $ followerId
+
+  followerInstruction followerId = do
+    model <- get
+    case Model.followerInstruction followerId allComponents model of
+      Left _ -> error "Failure"
```
Comment (dnadales) [inline ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs:175]: Maybe specify why we got this error.

**Review by dnadales — APPROVED** [review]
Great first step! LGTM.

File: ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs (hunk 6 above)
Comment (bartfrenk) [reply ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs:175]: Will be part of the next PR.

**Pull request merged**